component/azstorage/config.go

```diff
@@ -390,12 +390,26 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		az.stConfig.authConfig.SASKey = sanitizeSASKey(opt.SaSKey)
 	case EAuthType.MSI():
 		az.stConfig.authConfig.AuthMode = EAuthType.MSI()
-		if opt.ApplicationID == "" && opt.ResourceID == "" {
+		if opt.ApplicationID == "" && opt.ResourceID == "" && opt.ObjectID == "" {
```

Comment:
this should be or now right? and change the error string below

Reply:
no this is just saying if theyre all nil and we're msi then throw